core/dbt/include/global_project/macros/utils/array_construct.sql

```diff
@@ -0,0 +1,12 @@
+{% macro array_construct(inputs=[], data_type=api.Column.translate_type('integer')) -%}
+  {{ return(adapter.dispatch('array_construct', 'dbt')(inputs, data_type)) }}
+{%- endmacro %}
+
+{# all inputs must be the same data type to match postgres functionality #}
+{% macro default__array_construct(inputs, data_type) -%}
+    {% if inputs|length > 0 %}
+    array[ {{ inputs|join(' , ') }} ]
+    {% else %}
+    array[]::{{data_type}}[]
```

Comment:
looks like the adapter specific PRs don't have the functionality to return an empty array?

Reply:
This dbt-core PR includes multiple instances of `array_construct([])`, namely within these tests:
- `tests/adapter/dbt/tests/adapter/utils/fixture_array_construct.py`
- `tests/adapter/dbt/tests/adapter/utils/fixture_array_concat.py`
- `tests/adapter/dbt/tests/adapter/utils/test_array_append.py`

I was _hoping_ that these are sufficient to establish that the adapters handle the base case of an **empty array**. But if you take a look at those and it **doesn't** seem sufficient, then let's discuss it further to update the tests accordingly!

Reply:
as long as we're validating via tests that we don't need to implement the if condition for each adapter that's good to me